app/Models/Beatmap.php

```diff
@@ -279,6 +289,34 @@ public function getAttribute($key)
         };
     }
 
+    /**
+     * @return Collection<User>
+     */
+    public function getOwners(): Collection
+    {
+        $owners = $this->beatmapOwners->loadMissing('user')->map(
+            fn ($beatmapOwner) => $beatmapOwner->user ?? new DeletedUser(['user_id' => $beatmapOwner->user_id])
+        );
+
+        // TODO: remove when everything writes to beatmap_owners.
+        if (!$owners->contains(fn ($beatmapOwner) => $beatmapOwner->user_id === $this->user_id) && $this->user !== null) {
```

Comment:
related to the comment above, can't this be simplified to just `if ($owners->empty() && $this->user !== null) {`?

(there's also a potential weird bug/case already existing here)

Reply:
Is it a simplification if more states have to be considered instead of user is in or not? 🤨

Reply:
what more state? it's just whether there's anything in it or not

Reply:
It's also making another unnecessary assumption instead of just `user in or not`

Reply:
it skips looping through the array and so the check is way simpler

Reply:
also I think it's a fine assumption - there's either anything in the owners table or there's none

Reply:
It's also very pointless bike shedding

Reply:
more like optimization instead of bike shedding to me?